packages/system/src/use-style-config.ts

```diff
@@ -43,37 +43,35 @@ export function useStyleConfig(themeKey: any, props: any = {}, opts: any = {}) {
   type StylesRef = SystemStyleObject | Record<string, SystemStyleObject>
   const stylesRef = useRef<StylesRef>({})
 
-  return useMemo(() => {
-    if (styleConfig) {
-      const baseStyles = runIfFn(styleConfig.baseStyle ?? {}, mergedProps)
+  if (styleConfig) {
+    const baseStyles = runIfFn(styleConfig.baseStyle ?? {}, mergedProps)
 
-      const variants = runIfFn(
-        styleConfig.variants?.[mergedProps.variant] ?? {},
-        mergedProps,
-      )
+    const variants = runIfFn(
+      styleConfig.variants?.[mergedProps.variant] ?? {},
+      mergedProps,
+    )
 
-      const sizes = runIfFn(
-        styleConfig.sizes?.[mergedProps.size] ?? {},
-        mergedProps,
-      )
+    const sizes = runIfFn(
+      styleConfig.sizes?.[mergedProps.size] ?? {},
+      mergedProps,
+    )
 
-      const styles = mergeWith({}, baseStyles, sizes, variants)
+    const styles = mergeWith({}, baseStyles, sizes, variants)
 
-      if (opts?.isMultiPart && styleConfig.parts) {
-        styleConfig.parts.forEach((part: string) => {
-          styles[part] = styles[part] ?? {}
-        })
-      }
+    if (opts?.isMultiPart && styleConfig.parts) {
+      styleConfig.parts.forEach((part: string) => {
+        styles[part] = styles[part] ?? {}
+      })
+    }
 
-      const isStyleEqual = isEqual(stylesRef.current, styles)
+    const isStyleEqual = isEqual(stylesRef.current, styles)
 
-      if (!isStyleEqual) {
-        stylesRef.current = styles
-      }
+    if (!isStyleEqual) {
+      stylesRef.current = styles
     }
+  }
 
-    return stylesRef.current
-  }, [styleConfig, mergedProps, opts?.isMultiPart])
```

Comment:
`mergedProps` here is always a new object, preventing memoization

Reply:
Good catch!